File: 9-palindrome-number/palindrome-number.cs
using System;

public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int original = x;
        long rev = 0;

        while (x > 0) {
            rev = rev * 10 + x % 10;
            x /= 10;
        }

        return rev == original;
    }
}

File: 9-palindrome-number/palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 {
            return false;
        }

        let original = x as i64;
        let mut rev: i64 = 0;
        let mut x = x as i64;

        while x > 0 {
            rev = rev * 10 + x % 10;
            x /= 10;
        }

        rev == original
    }
}
